scripts/monitoring/cron-send-security-updates-count.py

```diff
@@ -0,0 +1,218 @@
+from configobj import ConfigObj
+#pylint: enable=import-error
+import glob
+import yaml
+
+ROOT = '/var/local/hostpkg'
+LOGFILE = '/var/log/cron-send-security-updates-count.log'
+
+import logging
+
+class SecurityUpdates(object):
+    '''
+      This is a check to see how many security updates are pending
+    '''
+    def __init__(self, options):
+        '''SecurityUpdateCount constructor'''
+        self.verbose = options.verbose
+        self.sec_updates = []
+        self.sec_advisory_whitelist = []
+        wlfile = '/host/etc/openshift_tools/security_advisory_whitelist.yml'
+        try:
+            with open(wlfile, 'r') as yaml_file:
+                try:
+                    tmp = yaml.load(yaml_file)
+                    if 'advisories' in tmp:
+                        self.sec_advisory_whitelist = tmp['advisories']
+                        if self.verbose:
+                            print('Whitelisted advisories: {0}'.format(','.join(self.sec_advisory_whitelist)))
+                    elif self.verbose:
+                        print('No advisories listed in {0}'.format(wlfile))
+                except yaml.YAMLError as err:
+                    print(err)
+        except IOError as err:
+            if self.verbose:
+                print('Unable to read {0}. Continuing.'.format(wlfile))
+                print(err)
+            logging.error('Unable to read %s. Continuing.\n%s', wlfile, err)
+
+    def get_security_updates(self):
+        '''Run yum to get a list of pending security updates'''
+        cmd = ['/usr/bin/yum', '--installroot='+ROOT, '-c', ROOT+'/etc/yum.conf',
+               '-q', '--security', 'updateinfo', 'list', 'updates']
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = proc.communicate()
+        if self.verbose:
+            print('stdout from yum:\n{0}'.format(stdout))
+            print('stderr from yum:\n{0}'.format(stderr))
+        if proc.returncode != 0:
+            self.sec_updates = []
+            raise subprocess.CalledProcessError(proc.returncode, ' '.join(cmd), output=stdout+stderr)
+        #lineslist = filter(lambda x: x and 'Errno' not in x, stdout.split('\n'))
+        #self.sec_updates = map(lambda x: dict(zip(['advisory', 'type', 'package'], re.split(r'\s+',x))), lineslist)
+        lineslist = [x for x in stdout.split('\n') if x and 'Errno' not in x]
+        self.sec_updates = [dict(zip(['advisory', 'type', 'package'], re.split(r'\s+', x))) for x in lineslist]
+        # now sec_updates looks like:
+        #     [ { 'advisory': 'RHSA-1999:1234', 'type': 'Important/Sec.', 'package': 'pkgname:1.2.3-0.el7.x86_64' },
+        #       { 'advisory': 'RHSA-2000:4321', 'type': 'Moderate/Sec.', 'package': 'somepkt-3:0.19.3-14.el7.x86_64' } ]
+
+        #for errline in filter(lambda x: x and 'Errno' in x, stdout.split('\n')):
+        for errline in [x for x in stdout.split('\n') if x and 'Errno' in x]:
```

Comment:
same as above with `line`.

Maybe be verbose that this is looking through errors.